ci/build_wheel.sh

```diff
@@ -10,30 +10,30 @@ source rapids-configure-sccache
 source rapids-date-string
 
 version=$(rapids-generate-version)
-commit=$(git rev-parse HEAD)
-
-RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
-
-# This is the version of the suffix with a preceding hyphen. It's used
-# everywhere except in the final wheel name.
-PACKAGE_CUDA_SUFFIX="-${RAPIDS_PY_CUDA_SUFFIX}"
-
-# Patch project metadata files to include the CUDA version suffix and version override.
-pyproject_file="${package_dir}/pyproject.toml"
-
-sed -i "s/name = \"${package_name}\"/name = \"${package_name}${PACKAGE_CUDA_SUFFIX}\"/g" ${pyproject_file}
 echo "${version}" > VERSION
-sed -i "/^__git_commit__/ s/= .*/= \"${commit}\"/g" "${package_dir}/${package_name}/_version.py"
 
-if [[ $PACKAGE_CUDA_SUFFIX == "-cu12" ]]; then
-    sed -i "s/cuda-python[<=>\.,0-9a]*/cuda-python>=12.0,<13.0a0/g" ${pyproject_file}
+if rapids-is-release-build; then
+    export RAPIDS_ONLY_RELEASE_DEPS=1
+fi
```

Comment:
I think before moving forward we should apply your suggestion from our previous meeting:
- Change the config option to `allow-nightly-deps`.
- Specify `allow-nightly-deps = 1` in the pyproject.toml in this PR
- Override that with the new environment variable `RAPIDS_ALLOW_NIGHTLY_DEPS=0` if `rapids-is-release-build` in this script.